### PR TITLE
Rebuild libxcb116 0 1 h351826 rework

### PR DESCRIPTION
**Pull request by croth1**
Try fixing the cross-compliation issues.

### CODE REVIEW

**Comment by croth1** [comment]
Seems not add the libraries during cross-compiling:

Cross compile: 
>024-06-22T11:13:02.0294032Z $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++  -fvisibility-inlines-hidden -fmessage-length=0 -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O3 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/gnuplot-5.4.8 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix  -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--allow-shlib-undefined -Wl,-rpath,$BUILD_PREFIX/lib -Wl,-rpath-link,$BUILD_PREFIX/lib -L$BUILD_PREFIX/lib -o gnuplot alloc.o axis.o breaders.o boundary.o color.o command.o contour.o datablock.o datafile.o dynarray.o encoding.o eval.o external.o fit.o gadgets.o getcolor.o graph3d.o graphics.o help.o hidden3d.o history.o internal.o interpol.o jitter.o libcerf.o matrix.o misc.o mouse.o multiplot.o parse.o plot.o plot2d.o plot3d.o pm3d.o readline.o save.o scanner.o set.o show.o specfun.o standard.o stats.o stdfn.o tables.o tabulate.o term.o time.o unset.o util.o util3d.o variable.o version.o voxelgrid.o vplot.o xdg.o     -lz    -ldl -lm 

vs. linux64 on linux64:
> 2024-06-22T11:11:11.0644973Z $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++  -fvisibility-inlines-hidden -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/gnuplot-5.4.8 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -fPIC   -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -L$PREFIX/lib  -o gnuplot alloc.o axis.o breaders.o boundary.o color.o command.o contour.o datablock.o datafile.o dynarray.o encoding.o eval.o external.o fit.o gadgets.o getcolor.o graph3d.o graphics.o help.o hidden3d.o history.o internal.o interpol.o jitter.o libcerf.o matrix.o misc.o mouse.o multiplot.o parse.o plot.o plot2d.o plot3d.o pm3d.o readline.o save.o scanner.o set.o show.o specfun.o standard.o stats.o stdfn.o tables.o tabulate.o term.o time.o unset.o util.o util3d.o variable.o version.o voxelgrid.o vplot.o xdg.o  wxterminal/gp_cairo.o wxterminal/gp_cairo_helpers.o  qtterminal/qt_term.o -lreadline  -lncurses  -lz -lgd -L$PREFIX/lib -lgd   -L$PREFIX/lib -lQt5Network -lQt5Svg -lQt5PrintSupport -lQt5Widgets -lQt5Gui -lQt5Core -ldl -lm  -L$PREFIX/lib -lz -lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lharfbuzz -lcairo -lglib-2.0 -liconv


@conda-forge/core do you think could be because smithy updated to a newer version of the toolchain? Seems strange that this would surface in the libxcb bump.

**Comment by hmaarrfk** [comment]
could we try to also remove the CDTs? It shouldn't be related, but should help troubleshoot things:

See 
https://github.com/conda-forge/qt-main-feedstock/pull/139/files

for many of the package transitions

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,22 +22,23 @@ requirements:
...
     - make  # [unix]
+    
+    - xorg-libxext  # [(build_platform != target_platform) and linux]
+    - xorg-libx11  # [(build_platform != target_platform) and linux]
+    - xorg-libxdamage  # [build_platform != target_platform]
+    - xorg-libxfixes  # [build_platform != target_platform]
+    - xorg-libxrender  # [build_platform != target_platform]
+    - xorg-libxt  # [build_platform != target_platform]
+    - xorg-libxxf86vm  # [build_platform != target_platform]
+
```
Comment (hmaarrfk) [inline recipe/meta.yaml:37]: are these dependencies here really needed?

Comment (croth1) [reply recipe/meta.yaml:37]: I... I don't really know - this is so old,  didn't even remember where I got the recipe from - probably migrated from bioconda loong, loooong time ago. I wanted to migrate over and fix the build before I start breaking it even more 😬

**Comment by hmaarrfk** [comment]
it might just be easier to build with emulation. sometimes cross compilation is too hard.

You can use `azure` as the build provider,

**Comment by croth1** [comment]
My current feeling is that [this line](https://github.com/conda-forge/gnuplot-feedstock/blob/main/recipe/build.sh#L53): `sed -ie 's/\(^LIBS.*\)/\1 -liconv/g' src/Makefile` was introduced for fixing a similar issue. @xhochy do you by any chance still recall why this was necessary? I still have no plausible explanation how this would pop up after rerendering and also curious that it used to build the native variant without this fix applied 🫤

**Comment by hmaarrfk** [comment]
you might just want to try and fix https://github.com/conda-forge/gnuplot-feedstock/pull/58 

my experience with cross compilation is that many upstream projects have gotten much better in recent years, so the issues might have "passed" with newer builds.

but great perseverance nonetheless

**Comment by croth1** [comment]
Haha, let me try a bit more - last time I checked the 6.x series had a known test failure that also would need patching. There's no free lunch here 😬

**Comment by croth1** [comment]
Ok, aarch64 cross compile actually worked this time. Not so sure about this one tho - bit suspicious that itÄ

>checking for powerpc64le-conda-linux-gnu-gcc... $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc
checking whether the C compiler works... no
configure: error: in `/home/conda/feedstock_root/build_artifacts/gnuplot_1719086215569/work/build-native':
configure: error: C compiler cannot create executables

 https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=960668&view=logs&j=8c139a40-380b-5518-bf6b-edcb18cbf319&t=e71f8d0d-13be-528e-b5e7-bfc20e0db699&l=1087
 
Interestingly the same worked for aarch. 
> checking for aarch64-conda-linux-gnu-gcc... $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc
checking whether the C compiler works... yes

Any clues welcome - maybe will have the time to do some detective work tomorrow.

File: recipe/build.sh [hunk 2]
```diff
@@ -6,27 +6,43 @@ cp $BUILD_PREFIX/share/gnuconfig/config.* .
...
+      libselinux-${cdt_name}-x86_64  \
+      xorg-x11-proto-devel-${cdt_name}-x86_64
+  fi
   (
     mkdir -p build-native
     pushd build-native
     LDFLAGS_FOR_BUILD=$(echo $LDFLAGS | sed "s?$PREFIX?$BUILD_PREFIX?g")
     LDFLAGS_LD_FOR_BUILD=$(echo $LDFLAGS_LD | sed "s?$PREFIX?$BUILD_PREFIX?g")
+    # CFLAGS -mcpu=power8 -mtune=power8 break the ppc64le build
+    CFLAGS_FOR_BUILD=$(echo $CFLAGS | sed -E 's?-mcpu=[^ ]+ ??; s?-mtune=[^ ]+ ??')
```
Comment (hmaarrfk) [inline recipe/build.sh:26]: :tada:

Comment (croth1) [reply recipe/build.sh:26]: yea slightly insane, I guess this needs a proper review 😁

**Comment by xhochy** [comment]
> My current feeling is that [this line](https://github.com/conda-forge/gnuplot-feedstock/blob/main/recipe/build.sh#L53): `sed -ie 's/\(^LIBS.*\)/\1 -liconv/g' src/Makefile` was introduced for fixing a similar issue. @xhochy do you by any chance still recall why this was necessary? I still have no plausible explanation how this would pop up after rerendering and also curious that it used to build the native variant without this fix applied 🫤

No idea here (anymore)

**Comment by croth1** [comment]
My current hypothesis is that somehow updating the compilers made configure take a wrong decision here:
>2024-06-23T19:22:52.2439519Z checking for library containing iconv_open... none required
But my motivation to figure out what exactly went wrong is rather limited 😄

**Pull request closed without merging**